bodhi/server/models.py

```diff
@@ -818,6 +808,31 @@ def _composite_karma(self):
 
         return positive_karma, negative_karma
 
+    def get_comments_since_karma_reset(self):
+        """
+        Return the comments since the most recent karma reset event, which
+        occurs whenever a build is added or removed from an Update.
+        :return: a list of comments since karma reset
+        :rtype: list
+        """
+        tempcomments = []
+        # We want to traverse the comments in reverse order so we only consider
+        # the most recent comments from any given user and only the comments
+        # since the most recent karma reset event.
+        for comment in reversed(self.comments):
+            if (comment.user.name == u'bodhi' and
+                    ('New build' in comment.text or 'Removed build' in comment.text)):
+                # We only want to consider comments since the most recent karma
+                # reset, which happens whenever a build is added or removed
+                # from an Update. Since we are traversing the comments in
+                # reverse order, once we find one of these comments we can
+                # simply exit this loop.
+                break
+            else:
+                tempcomments.append(comment)
```

Comment:
If you use Python's [```yield``` statement](https://docs.python.org/2/reference/simple_stmts.html#yield) in this method, it will become a [generator](https://wiki.python.org/moin/Generators) and will not have to load all of the comments into a temporary list. This will make the code a bit more efficient, nearly as efficient as it had been before (though incorrect before ☺). Also, you do not need the ```else``` clause because not hitting the ```break``` implies that the ```if``` statement wasn't matched. Thus, I recommend refactoring the above to something like this:

```
for comment in reversed(self.comments):
    if (comment.user.name == u'bodhi' and
            ('New build' in comment.text or 'Removed build' in comment.text)):
    yield comment
```